databricks/koalas/indexing.py

```diff
@@ -400,16 +399,39 @@ def _select_rows(self, rows_sel):
                 # If slice is None - select everything, so nothing to do
                 return None, None
             elif len(self._internal.index_columns) == 1:
+                sdf = self._kdf_or_kser._internal.sdf
+                index_column = self._kdf_or_kser.index.to_series()
+                index_data_type = index_column.spark_type
                 start = rows_sel.start
                 stop = rows_sel.stop
 
-                index_column = self._kdf_or_kser.index.to_series()
-                index_data_type = index_column.spark_type
+                # get natural order from '__natural_order__' from start to stop
+                # to keep natural order when type of rows are StringType
+                if isinstance(index_data_type, StringType):
+                    start_and_stop = (
+                        sdf.select(index_column._scol, NATURAL_ORDER_COLUMN_NAME)
+                           .where((index_column._scol == start) | (index_column._scol == stop))
+                           .collect())
+
+                    start = [row[1] for row in start_and_stop if row[0] == start]
+                    start = start[0] if len(start) > 0 else None
+
+                    stop = [row[1] for row in start_and_stop if row[0] == stop]
+                    stop = stop[0] if len(stop) > 0 else None
+
+                    order_column = sdf[NATURAL_ORDER_COLUMN_NAME]
+                    if start is None and rows_sel.start is not None:
+                        raise KeyError(rows_sel.start)
+                    if stop is None and rows_sel.stop is not None:
+                        raise KeyError(rows_sel.stop)
+                    index_data_type = LongType()
+                else:
+                    order_column = index_column._scol
```

Comment:
@itholic, I think we should always use the natural order regardless of types. For example:

```python
>>> pd.DataFrame({'a': [1,2,3]}, index=[3,2,1]).loc[3:2, 'a']
3    1
2    2
```

but

```python
>>> ks.DataFrame({'a': [1,2,3]}, index=[3,2,1]).loc[3:2, 'a']
Series([], Name: a, dtype: int64)
```

Reply:
ah, i missed that case, let me manage it soon
thanks !

Reply:
i've just fixed :)